src/lib/Preview.js

```diff
@@ -148,21 +148,28 @@ class Preview extends EventEmitter {
     /**
      * Primary function for showing a preview of a file.
      *
-     * @param {string} fileId - Box File ID
+     * @param {string|Object} fileIdOrFile - Box File ID or well-formed Box File object
      * @param {string|Function} token - Access token string or generator function
      * @param {Object} [options] - Optional preview options
      * @return {void}
      */
-    show(fileId, token, options = {}) {
+    show(fileIdOrFile, token, options = {}) {
         // Save a reference to the options to be used later
         if (typeof token === 'string' || typeof token === 'function') {
             this.previewOptions = Object.assign({}, options, { token });
         } else {
             throw new Error('Missing access token!');
         }
 
-        // load the preview
-        this.load(fileId);
+        // Check if passed in file is a well-formed Box File object
+        if (typeof fileIdOrFile === 'object' && !checkFileValid(fileIdOrFile)) {
```

Comment:
Should move the === object check inside `checkFileValid()` since it already has a similar null check